.github/workflows/release.yml

```diff
@@ -28,22 +28,36 @@ jobs:
 
       - name: Build binary
         run: |
+          VERSION=${{ github.ref_name }}
+          COMMIT=${{ github.sha }}
+          BUILD_TIME=$(date -u +%Y%m%d.%H%M%S)
+          BRANCH=${GITHUB_REF#refs/heads/}
+          
+          OUTPUT_NAME="gci-${{ matrix.os }}-${{ matrix.arch }}"
+          if [ "${{ matrix.os }}" = "windows" ]; then
+            OUTPUT_NAME="${OUTPUT_NAME}.exe"
+          fi
+          
           GOOS=${{ matrix.os }} GOARCH=${{ matrix.arch }} \
-          go build -ldflags="-w -s -X main.Version=${{ github.ref_name }} -X main.BuildTime=$(date -u +%Y%m%d.%H%M%S)" \
-          -o git-ci-${{ matrix.os }}-${{ matrix.arch }} \
-          ./cmd
+          go build -ldflags="-w -s \
+            -X main.Version=${VERSION} \
+            -X main.Commit=${COMMIT} \
+            -X main.BuildTime=${BUILD_TIME} \
+            -X main.Branch=${BRANCH}" \
+          -o ${OUTPUT_NAME} ./cmd
 
       - name: Upload artifact
         uses: actions/upload-artifact@v4
         with:
-          name: git-ci-${{ matrix.os }}-${{ matrix.arch }}
-          path: git-ci-${{ matrix.os }}-${{ matrix.arch }}
+          name: gci-${{ matrix.os }}-${{ matrix.arch }}
+          path: gci-${{ matrix.os }}-${{ matrix.arch }}${{ matrix.os == 'windows' && '.exe' || '' }}
 
   release:
     name: Release
     runs-on: ubuntu-latest
     needs: [build]
-    if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Out of curiosity, why removing this conditional check ?
I add it to run the release job only from a refs/tags